vertx-web-client/src/main/java/io/vertx/ext/web/client/cache/CacheManager.java

```diff
@@ -0,0 +1,38 @@
+package io.vertx.ext.web.client.cache;
+
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
+
+import java.util.Optional;
+
+/**
+ * Abstracts caching of HTTP requests and responses
+ * @author Alexey Soshin
+ */
+public interface CacheManager {
```

Comment:
the cache key should be computed by the interceptor and not the CacheManager implementation